src/Platform/Microsoft.Testing.Extensions.TrxReport/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// Based on the XXH128 implementation from https://github.com/Cyan4973/xxHash.
+using System.Buffers.Binary;
+using System.Numerics;
+
+using static System.IO.Hashing.XxHashShared;
+
+#pragma warning disable RS0030 // Do not use banned APIs - Debug is okay here. RoslynDebug isn't yet available in PlatformServices which links this file.
+
+namespace System.IO.Hashing;
+
+/// <summary>Provides an implementation of the XXH128 hash algorithm for generating a 128-bit hash.</summary>
+/// <remarks>
+/// For methods that persist the computed numerical hash value as bytes,
+/// the value is written in the Big Endian byte order.
+/// </remarks>
+#if NET
+[SkipLocalsInit]
+#endif
+public sealed unsafe class XxHash128
```

Comment:
Should this be internal so that it does not collide with the official implementation?